

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Setting up configuration file &mdash; FitHiChIP 0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../static/documentation_options.js"></script>
        <script type="text/javascript" src="../static/jquery.js"></script>
        <script type="text/javascript" src="../static/underscore.js"></script>
        <script type="text/javascript" src="../static/doctools.js"></script>
        <script type="text/javascript" src="../static/language_data.js"></script>
    
    <script type="text/javascript" src="../static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Details of FitHiChIP outputs" href="output.html" />
    <link rel="prev" title="Quick start" href="quickstart.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> FitHiChIP
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick start</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Setting up configuration file</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#a-support-of-input-file-formats">A) Support of input file formats</a></li>
<li class="toctree-l2"><a class="reference internal" href="#b-other-important-parameters">B) Other important parameters</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Details of FitHiChIP outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffLoops.html">Differential analysis of HiChIP loops</a></li>
<li class="toctree-l1"><a class="reference internal" href="Utilities.html">Various utility scripts / functions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked question</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">FitHiChIP</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Setting up configuration file</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../sources/usage/configuration.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="setting-up-configuration-file">
<h1>Setting up configuration file<a class="headerlink" href="#setting-up-configuration-file" title="Permalink to this headline">¶</a></h1>
<p>As mentioned in the Quick Start module, FitHiChIP is executed by typing the following command in a bash terminal (assuming the executable is in current directory):</p>
<blockquote>
<div>sh FitHiChIP_HiCPro.sh -C configuration_file_name</div></blockquote>
<p>Here we mention the parameters and recommended values to be provided in the configuration file. Each entry of the configuration file has the following format:</p>
<blockquote>
<div>Param=ParamValue</div></blockquote>
<p>where,</p>
<blockquote>
<div><ul class="simple">
<li>“Param” indicates one parameter (variable)</li>
<li>“ParamValue” is the corresponding value (numeric or string format).</li>
</ul>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">We recommend users to mention absolute paths of files / folders in the configuration file.</p>
</div>
<div class="section" id="a-support-of-input-file-formats">
<h2>A) Support of input file formats<a class="headerlink" href="#a-support-of-input-file-formats" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p>FitHiChIP supports input interaction file in <strong>either of three possible options</strong>:</p>
<ol class="arabic simple">
<li>Either specify the parameter A.1</li>
<li>Or, specify both the parameters A.2 and A.3</li>
<li>or specify the parameter A.4</li>
</ol>
<p>Details of these parameters are:</p>
<p>A.1) ValidPairs:</p>
<blockquote>
<div><ul class="simple">
<li>Valid pairs generated by HiC-pro pipeline (either <strong>simple text</strong> format, or can be in <strong>gzipped</strong> format).</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Usually, name of the valid pairs file generated from HiC-pro pipeline is <strong>${HICPRODIR}/hic_results/data/rawdata/rawdata.allValidPairs</strong> where <strong>${HICPRODIR}</strong> is the directory containing HiC-pro output.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Currently, chromosome information within the valid pairs file need to be alphanumeric, i.e. chr1, chr2, etc. If the validpairs file contains numeric chromosome information (like 1, 2, etc), they need to be replaced with chr1, chr2, etc.</li>
</ul>
</div>
</div></blockquote>
<p>A.2) Interval:</p>
<blockquote>
<div><ul class="simple">
<li>File depicting the bins / intervals of the interaction matrix. Generated by applying HiC-pro pipeline utility function <strong>build_matrix</strong> on the valid pairs file.</li>
<li>Size of an interval depends on the bin size. Individual bins are also assigned a distinct number.</li>
<li>By default, name of this file ends with a suffix <strong>_abs.bed</strong></li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Chromosome information need to be alphanumeric (as mentioned in the parameter A.1).</li>
</ul>
</div>
</div></blockquote>
<p>A.3) Matrix:</p>
<blockquote>
<div><ul class="simple">
<li>File listing the number of interactions (contacts) among the bins listed in the <strong>Interval</strong> (parameter A.2) file.</li>
<li>The file has three columns: 1) first interacting bin number, 2) second interacting bin number, 3) contact count between those two bins.</li>
<li>Generated by applying HiC-pro pipeline utility function <strong>build_matrix</strong> on the valid pairs file.</li>
</ul>
</div></blockquote>
<p>A.4) Bed:</p>
<blockquote>
<div><ul class="simple">
<li>Tab delimited text file containing the locus pairs (binned) and the contact count.</li>
<li>Of the format:        chr1    start1          end1    chr2    start2          end2    contact</li>
<li>Can be generated from any HiC or HiChIP data processing pipelines, or custom scripts.</li>
</ul>
</div></blockquote>
</div></blockquote>
</div>
<div class="section" id="b-other-important-parameters">
<h2>B) Other important parameters<a class="headerlink" href="#b-other-important-parameters" title="Permalink to this headline">¶</a></h2>
<blockquote>
<div><p>With the input file provided by one of the above mentioned options, user should also provide the following parameters:</p>
<p>B.1) PeakFile:</p>
<blockquote>
<div><ul class="simple">
<li>Reference ChIP-seq or HiChIP peak file.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This is a mandatory parameter if user is analyzing HiChIP data and <strong>IntType</strong> (parameter (B.4)) is 1, 2, 3 or 5.</p>
</div>
<ul class="simple">
<li>User may use pre-computed ChIP-seq peaks in ENCODE (<a class="reference external" href="https://www.encodeproject.org/">https://www.encodeproject.org/</a>),</li>
<li>or, user may compute HiChIP peaks and use them (described in the page <a class="reference internal" href="Utilities.html#various-utility-scripts-functions"><span class="std std-ref">Various utility scripts / functions</span></a>).</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>user may employ <strong>macs2</strong>, or a recent package <strong>hichipper</strong> (Lareau et al 2018) to compute the peaks from HiChIP data, and provide the bed formatted peak file as an input to FitHiChIP.</li>
<li>If MACS2 is employed, <a href="#id1"><span class="problematic" id="id2">*</span></a>.narrowPeak formatted peak file is to be provided.</li>
</ol>
</div>
</div></blockquote>
<p>B.2) OutDir:</p>
<blockquote>
<div><ul class="simple">
<li>Output directory which would contain all the results.</li>
<li>Default: present working directory.</li>
</ul>
</div></blockquote>
<p>B.3) ChrSizeFile:</p>
<blockquote>
<div><ul class="simple">
<li>File containing the chromosome size corresponding to the reference genome. <strong>Mandatory parameter.</strong></li>
<li>For example, the file <strong>chrom_hg19.sizes</strong> within the folder “TestData” is applicable for the reference genome hg19.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>For example, chromosome size file for the reference genome hg38 can be obtained from the link: <a class="reference external" href="https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/hg38.chrom.sizes">https://hgdownload.cse.ucsc.edu/goldenpath/hg38/bigZips/hg38.chrom.sizes</a></li>
<li>Another option is to use the utility “fetchChromSizes” from UCSC (<a class="reference external" href="https://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/fetchChromSizes">https://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/fetchChromSizes</a>) and specify the reference genome information, to download corresponding chromosome size file.</li>
</ol>
</div>
</div></blockquote>
<p>B.4) IntType:</p>
<blockquote>
<div><p>Type of interaction (foreground) reported by FitHiChIP. Options are:</p>
<ol class="arabic simple">
<li><strong>peak to peak</strong>: contacts between all pairs of peak segments (subject to fixed size binning)</li>
<li><strong>peak to non peak</strong>: contacts between pairs of segments such that one is a peak and the other is a non-peak segment</li>
<li><strong>peak to all (default)</strong>: here one interacting segment is a peak, while the other can be a peak or a non-peak. Thus, encapsulates options 1 and 2.</li>
<li><strong>all to all</strong>: Interactions between every possible pairs of segments, similar to Hi-C.</li>
<li>Everything from 1 to 4. That is, all of the above mentioned interactions are computed.</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Default is 3, i.e. peak-to-all (both peak-to-peak and peak-to-nonpeak interactions are reported by FitHiChIP)</p>
</div>
</div></blockquote>
<p>B.5) BINSIZE:</p>
<blockquote>
<div>Size of the bins, depicting the resolution employed. Default= 5000 (means 5 Kb resolution)</div></blockquote>
<p>B.6) LowDistThr:</p>
<blockquote>
<div>Lower distance threshold of interaction between two intervals (CIS). Default: 20000 (indicates 20 Kb). Interactions below this distance threshold will not be considered for statistical significance.</div></blockquote>
<p>B.7) UppDistThr:</p>
<blockquote>
<div>Upper distance threshold of interaction between two intervals (CIS). Default: 2000000 (indicates 2 Mb). Interactions above this distance threshold will not be considered for statistical significance.</div></blockquote>
<p>B.8) QVALUE:</p>
<blockquote>
<div>Minimum FDR (q-value) cutoff for detecting significant interactions. Default: 0.01</div></blockquote>
<p>B.9) UseP2PBackgrnd:</p>
<blockquote>
<div><p>Can be 0 or 1. Applicable only for peak to all interactions (IntType = 3).</p>
<blockquote>
<div><ul class="simple">
<li>1: peak to peak (or stringent) background (locus pairs) are used for contact probability estimation. Refers to FitHiChIP(S)</li>
<li>0: peak to all (or loose) background (locus pairs) are used for contact probability estimation. Refers to FitHiChIP(L)</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">We recommend users to execute FitHiChIP using both loose and stringent settings. Specifically, for low to moderate sequencing depth, use 0, whereas for very high sequencing depth, employ a value 1.</p>
</div>
</div></blockquote>
</div></blockquote>
<p>B.10) BiasType:</p>
<blockquote>
<div><p>Can be 1 or 2. Indicates the type of bias correction used.</p>
<blockquote>
<div><ul class="simple">
<li>A value of 1 means that coverage bias regression is used (default).</li>
<li>Value of 2 means that ICE bias regression is computed.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Results in the manuscript employ BiasType=1</p>
</div>
</div></blockquote>
</div></blockquote>
<p>B.11) MergeInt:</p>
<blockquote>
<div><ul class="simple">
<li>Has the value 0 or 1.</li>
<li>If 1, merge filtering (+M) is enabled. Otherwise (0) not.</li>
<li><strong>We recommend setting this value as 1.</strong></li>
</ul>
</div></blockquote>
<p>B.12) PREFIX:</p>
<blockquote>
<div>Prefix string used before any output file name. Default = “FitHiChIP”</div></blockquote>
<p>B.13) OverWrite:</p>
<blockquote>
<div><p>A binary variable (1/0).</p>
<blockquote>
<div><ul class="simple">
<li>If 1, overwrites existing FitHiChIP output files.</li>
<li>Otherwise (0) skips re-computation of existing outputs.</li>
</ul>
</div></blockquote>
</div></blockquote>
</div></blockquote>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="output.html" class="btn btn-neutral float-right" title="Details of FitHiChIP outputs" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="quickstart.html" class="btn btn-neutral float-left" title="Quick start" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Sourya Bhattacharyya, Ferhat Ay

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>