

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Details of FitHiChIP outputs &mdash; FitHiChIP 0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../static/documentation_options.js"></script>
        <script type="text/javascript" src="../static/jquery.js"></script>
        <script type="text/javascript" src="../static/underscore.js"></script>
        <script type="text/javascript" src="../static/doctools.js"></script>
        <script type="text/javascript" src="../static/language_data.js"></script>
    
    <script type="text/javascript" src="../static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Differential analysis of HiChIP loops" href="DiffLoops.html" />
    <link rel="prev" title="Setting up configuration file" href="configuration.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> FitHiChIP
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick start</a></li>
<li class="toctree-l1"><a class="reference internal" href="configuration.html">Setting up configuration file</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Details of FitHiChIP outputs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#summary-html-file">Summary HTML file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#details-of-output-files">Details of output files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#details-of-fithichip-significant-interactions">Details of FitHiChIP significant interactions</a></li>
<li class="toctree-l2"><a class="reference internal" href="#details-of-merge-filtering-output">Details of merge filtering output</a></li>
<li class="toctree-l2"><a class="reference internal" href="#visualizing-significant-interactions-in-washu-epigenome-browser">Visualizing significant interactions in WashU epigenome browser</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#using-current-version">Using current version</a></li>
<li class="toctree-l3"><a class="reference internal" href="#using-legacy-version">Using legacy version</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#visualizing-significant-interactions-in-ucsc-genome-browser">Visualizing significant interactions in UCSC genome browser</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="DiffLoops.html">Differential analysis of HiChIP loops</a></li>
<li class="toctree-l1"><a class="reference internal" href="Utilities.html">Various utility scripts / functions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked question</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">FitHiChIP</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Details of FitHiChIP outputs</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../sources/usage/output.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="details-of-fithichip-outputs">
<h1>Details of FitHiChIP outputs<a class="headerlink" href="#details-of-fithichip-outputs" title="Permalink to this headline">¶</a></h1>
<div class="section" id="summary-html-file">
<h2>Summary HTML file<a class="headerlink" href="#summary-html-file" title="Permalink to this headline">¶</a></h2>
<p>Successful execution of FitHiChIP pipeline generates an HTML file <strong>Summary_results_FitHiChIP.html</strong> within the output directory <strong>OutDir</strong> specified in the configuration file. This file lists the output files containing significant interactions from the FitHiChIP pipeline, according to the given input parameters.</p>
</div>
<div class="section" id="details-of-output-files">
<h2>Details of output files<a class="headerlink" href="#details-of-output-files" title="Permalink to this headline">¶</a></h2>
<p>Here we describe all the files and folders existing within the specified output directory <strong>OutDir</strong>.</p>
<p>User should additionally open this page <a class="reference internal" href="configuration.html#setting-up-configuration-file"><span class="std std-ref">Setting up configuration file</span></a> to tally with the configuration parameters:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first"><strong>Parameters.txt</strong>: file listing the parameters used.</p>
</li>
<li><p class="first"><strong>HiCPro_Matrix_BinSize”BINSIZE”</strong>:</p>
<blockquote>
<div><p>Where “BINSIZE” is specified in the parameter (B.5).</p>
<p>If the parameters (A.2) and (A.3) are empty, this directory contains the interaction matrix files (named as <em>MatrixHiCPro_abs.bed</em> and <em>MatrixHiCPro.matrix</em>) generated from the input valid pairs file (parameter A.1).</p>
<ul>
<li><p class="first"><strong>*.interactions.initial.bed</strong>: CIS locus pairs along with their contact counts, corresponding to the input valid pairs (or contact matrix).</p>
</li>
<li><p class="first"><strong>L_”l”_U”u”</strong>: folder name according to the distance thresholds “l” and “u” specified in the parameters (B.6) and (B.7).</p>
<blockquote>
<div><ul class="simple">
<li><strong>*.cis.interactions.DistThr.bed</strong>: File containing CIS interactions within the distance range from “l” to “u”.</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><strong>NormFeatures</strong>:</p>
<blockquote>
<div><p>Folder containing various statistical features computed for individual genomic intervals / bins (with respect to the specified “BINSIZE”).</p>
<ul>
<li><p class="first">User should specifically look at the file <strong>*.AllBin_CompleteFeat.bed</strong>, which has the following format:</p>
<blockquote>
<div><ul class="simple">
<li>Columns 1 to 3: chromosome bin.</li>
<li>Column 4: Coverage of this bin (number of reads mapped onto it)</li>
<li>Column 5: if 1, this bin is a <em>peak-bin</em> (overlapping with the reference peaks, provided in the parameter B.1); otherwise (0) <em>non-peak bin</em>.</li>
<li>Column 6: Bias value of a bin; either coverage or ICE bias depending on the parameter B.10</li>
<li>Columns 7 and 8: unused columns.</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><strong>FitHiChIP_”INTTYPE”_b”BINSIZE”_L”l”_U”u”</strong></p>
<blockquote>
<div><p>Folder according to the specified foreground (type of interactions to be reported; parameter B.4), bin size (parameter B.5), and the distance thresholds (parameters B.6 and B.7)</p>
<ul>
<li><p class="first">INTTYPE is a string with one of the following values:</p>
<blockquote>
<div><ul class="simple">
<li>ALL2ALL (when the parameter B.4 = 4)</li>
<li>Peak2Peak (parameter B.4 = 1),</li>
<li>Peak2NonPeak (parameter B.4 = 2), or</li>
<li>Peak2ALL (parameter B.4 = 3).</li>
</ul>
</div></blockquote>
</li>
<li><p class="first">Within this folder, a directory structure of the following name (pattern) is created:</p>
<blockquote>
<div><p><strong>P2PBckgr_[UseP2PBackgrnd]/[BiasType]</strong></p>
</div></blockquote>
</li>
</ul>
<p>Where:</p>
<blockquote>
<div><ul class="simple">
<li>[UseP2PBackgrnd]: Value of the parameter (B.9) - 0 (loose background) or 1 (stringent background).</li>
<li>[BiasType]: A string, either “Coverage_Bias” or “ICE_Bias”, according to the parameter (B.10).</li>
</ul>
</div></blockquote>
<ul>
<li><p class="first">Within this directory, following files are important:</p>
<blockquote>
<div><ul>
<li><p class="first"><strong>Interactions.sortedGenDist.bed</strong>: Interactions sorted by increasing genomic distance. Used as the input to FitHiChIP.</p>
</li>
<li><p class="first"><strong>FitHiC_BiasCorr</strong>: Directory containing the significant interactions:</p>
<blockquote>
<div><ul>
<li><p class="first"><strong>Bin_Info.log</strong>: Equal occupancy bins employed in FitHiChIP</p>
</li>
<li><p class="first"><strong>configuration.txt</strong>: list of configuration parameters for calling significant interactions from FitHiChIP</p>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC.bed</strong>: Lists all input interactions along with their significance (p-value and FDR) computed using FitHiChIP.</p>
<blockquote>
<div><ul class="simple">
<li>Value of <strong>PREFIX</strong> is provided in the configuration parameter (B.12)</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last"><strong>Important:</strong> For differential analysis of HiChIP loops (page <a class="reference internal" href="DiffLoops.html#differential-analysis-of-hichip-loops"><span class="std std-ref">Differential analysis of HiChIP loops</span></a>), this file (containing all interactions along with FitHiChIP significance values) is to be provided as input.</p>
</div>
</div></blockquote>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”.bed</strong>:</p>
<blockquote>
<div><p>FitHiChIP generated set of significant interactions according to the q-value threshold (QVALUE) specified in parameter (B.8).</p>
<p>For example, if QVALUE=0.01 (default), the file name becomes <strong>*PREFIX*.interactions_FitHiC_Q0.01.bed</strong>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li><strong>Important:</strong> This file is the set of significant interactions generated by FitHiChIP, without any merge filtering.</li>
<li>For peak-to-all foreground (default output mode), if <strong>UseP2PBackgrnd</strong> = 0, then these loops refer to FitHiChIP(L) output.</li>
<li>If <strong>UseP2PBackgrnd</strong> = 1, these loops refer to FitHiChIP(S) output.</li>
</ol>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>If user requires loops filtered by a FDR threshold different from the parameter QVALUE (say the custom threshold is 0.05 and the QVALUE is 0.01), user may employ the following awk script to generate the significant interactions.</p>
<blockquote class="last">
<div><p><strong>awk ‘$NF&lt;0.05’ *PREFIX*.interactions_FitHiC.bed &gt; FitHiChIP_out_Q0.05.bed</strong></p>
</div></blockquote>
</div>
</div></blockquote>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”_WashU.bed.gz.*</strong>:</p>
<blockquote>
<div><p>Significant FitHiChIP loops (<strong>*PREFIX*.interactions_FitHiC_Q”qval”.bed</strong>) converted to WashU epigenome browser compatible format.</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Merge_Nearby_Interactions</strong>:</p>
<blockquote>
<div><p>Directory containing the output of merge filtering operation, if the parameter (B.11) = 1.</p>
<ul>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”_MergeNearContacts.bed</strong>:</p>
<blockquote>
<div><p>FitHiChIP significant loops returned after the merge filtering operation.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>For peak-to-all foreground (default output mode), if <strong>UseP2PBackgrnd</strong> = 0, this set of loops refer to FitHiChIP(L+M) output.</li>
<li>If <strong>UseP2PBackgrnd</strong> = 1, this set of loops refer to FitHiChIP(S+M) output.</li>
</ol>
</div>
</div></blockquote>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”_MergeNearContacts_WashU.bed.gz.*</strong>:</p>
<blockquote>
<div><p>Significant FitHiChIP loops (<strong>*PREFIX*.interactions_FitHiC_Q”qval”_MergeNearContacts.bed</strong>) converted to WashU epigenome browser compatible format.</p>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="details-of-fithichip-significant-interactions">
<h2>Details of FitHiChIP significant interactions<a class="headerlink" href="#details-of-fithichip-significant-interactions" title="Permalink to this headline">¶</a></h2>
<p>As mentioned above, the file <strong>*PREFIX*.interactions_FitHiC.bed</strong> contains all the loops and their significance values. The file <strong>*PREFIX*.interactions_FitHiC_Q”qval”.bed</strong> contains the significant interactions with respect to the specified FDR threshold <strong>qval</strong></p>
<p>Both of these files contain the following fields:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Fields 1 to 6 denote the interacting bins for a given interaction.</p>
</li>
<li><p class="first">Field 7 is the contact count.</p>
</li>
<li><p class="first">Fields 8 to 13 show various properties of the first interacting bin. This include the HiChIP coverage, whether the bin overlaps with ChIP-seq peak (isPeak), bias (coverage or ICE depending on the bias parameter), mappability, GC content, and number of restriction sites.</p>
<blockquote>
<div><p>Currently, the values of mappability, GC content, and number of restriction sites are set as 0. We did not remove the fields though, to keep the same output format.</p>
</div></blockquote>
</li>
<li><p class="first">Fields 14 to 19 show the respective properties for the second interacting bin.</p>
</li>
<li><p class="first">Field 20 (p) denotes the spline fit probability.</p>
</li>
<li><p class="first">Field 21 (exp_cc_Bias) is the expected contact count of the current interaction, after applying bias regression model.</p>
</li>
<li><p class="first">Fields 22 and 23 denote the probability computed from the “exp_cc_Bias” value, and binomial distribution based refined probability value.</p>
</li>
<li><p class="first">Fields 24 and 25 show the final FitHiChiP p-value and q-value of the current interaction.</p>
</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="details-of-merge-filtering-output">
<h2>Details of merge filtering output<a class="headerlink" href="#details-of-merge-filtering-output" title="Permalink to this headline">¶</a></h2>
<p>As mentioned above, the file <strong>*PREFIX*.interactions_FitHiC_Q”qval”_MergeNearContacts.bed</strong> within the directory <strong>Merge_Nearby_Interactions</strong> contains the significant loops returned after merge filtering operation.</p>
<p>This file contains the following fields:</p>
<blockquote>
<div><ol class="arabic simple">
<li>First 6 fields are the interacting bins corresponding to the retained set of interactions (after merge filtering).</li>
<li>Next 3 fields denote the contact count, p-value and q-value of the retained interactions.</li>
<li>bin1_low and bin1_high denotes the lower and upper coordinates of the first set of interacting bins (x axis of the proposed connected component model). For example, if the set of bins (x-1, y-1), (x, y) and (x+1,y+1) are provided as inputs, bin1_low becomes (x-1) * BINSIZE while bin1_high becomes (x+1) * BINSIZE.</li>
<li>bin2_low and bin2_high, denotes the lower and upper coordinates of the second set of interacting bins. Considering the previous example, bin2_low = (y-1) * BINSIZE and bin2_high = (y+1) * BINSIZE</li>
<li>sumCC denotes the sum of contact counts of all the significant interactions applied to the connected component model, with respect to the current retained interaction. Considering the above example, sumCC would be equal to the contact counts of interactions (x-1, y-1), (x, y) and (x+1,y+1).</li>
<li>StrongConn is between 0 and 1, and denotes the fraction of all possible interactions within the connected component which are significant. Considering the above example, the connected component enclosing the bins (x-1), x, and (x+1) at one end, and the bins (y-1), y, and (y+1) at the other end, can have at most 9 interactions, out of which 3 are significant. So, StrongConn is 0.33 in this case. Higher value of StrongConn indicates dense connected region.</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="visualizing-significant-interactions-in-washu-epigenome-browser">
<h2>Visualizing significant interactions in WashU epigenome browser<a class="headerlink" href="#visualizing-significant-interactions-in-washu-epigenome-browser" title="Permalink to this headline">¶</a></h2>
<p>FitHiChIP interactions can be readily visualized in the WashU epigenome browser:</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>Two different versions of WashU epigenome browser currently exist:</p>
<blockquote>
<div><p><strong>current</strong> (<a class="reference external" href="http://epigenomegateway.wustl.edu/browser/">http://epigenomegateway.wustl.edu/browser/</a>) and</p>
<p><strong>legacy</strong> (<a class="reference external" href="http://epigenomegateway.wustl.edu/legacy/">http://epigenomegateway.wustl.edu/legacy/</a>).</p>
</div></blockquote>
<p class="last">These two versions have substantial differences - file formats supported in these two versions are also different. So, we describe using both of these versions.</p>
</div>
<div class="section" id="using-current-version">
<h3>Using current version<a class="headerlink" href="#using-current-version" title="Permalink to this headline">¶</a></h3>
<p>The files <strong>*_WashU.bed.gz</strong> mentioned above are all compatible with the <strong>current</strong> version. User can visualize these tracks by step by step execution of the following procedure:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Open <a class="reference external" href="http://epigenomegateway.wustl.edu/browser/">http://epigenomegateway.wustl.edu/browser/</a> in a browser.</li>
<li>Select reference genome. For the test data, human (hg19) is the reference genome.</li>
<li>select <strong>Tracks</strong> -&gt; <strong>upload Local Track</strong></li>
<li>Choose track file type: <strong>longrange</strong></li>
<li>Choose track file: select both <strong>*_WashU.bed.gz</strong> and <strong>*_WashU.bed.gz.tbi</strong> files <strong>together (2 files)</strong>.</li>
<li>Close the track upload box (click at the (X) sign in the right corner)</li>
<li>Now the track is displayed in the browser window. Right click on the track name, and select <strong>DISPLAY MODE</strong> as <strong>ARC</strong></li>
</ol>
</div></blockquote>
</div>
<div class="section" id="using-legacy-version">
<h3>Using legacy version<a class="headerlink" href="#using-legacy-version" title="Permalink to this headline">¶</a></h3>
<p>User needs to convert the FitHiChIP significant loops (in .bed format, not gzipped version) as the following:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Output of <strong>FitHiChIP(L)</strong> or <strong>FitHiChIP(S)</strong> - file <strong>FitHiC_Q”qval”.bed</strong></p>
<blockquote>
<div><p>Let us denote the file as <em>inp.bed</em>. It contains a header line. First 6 fields denote the interacting bins and the last field denotes the q-value. User needs to apply the <strong>following awk script</strong> to this file:</p>
<blockquote>
<div><p>awk -F[‘\t’] ‘{if (NR &gt; 1) {if ($NF &gt; 0) {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”\t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”\t”(-log($NF)/log(10))} else {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”\t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”\t500”}}}’ <em>inp.bed</em> &gt; <em>out_WashU.bed</em></p>
</div></blockquote>
</div></blockquote>
</li>
<li><p class="first">Output of <strong>FitHiChIP(L+M)</strong> or <strong>FitHiChIP(S+M)</strong> -  file <strong>FitHiC_Q”qval”_MergeNearContacts.bed</strong></p>
<blockquote>
<div><p>Let us denote the file as <em>inp_merged.bed</em>. It contains a header line. First 6 fields denote the interacting bins and the 9th field stores the q-value. User needs to apply the following awk script to this file:</p>
<p>awk -F[‘\t’] ‘{if (NR &gt; 1) {if ($9 &gt; 0) {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”\t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”\t”(-log($9)/log(10))} else {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”\t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”\t500”}}}’ <em>inp_merged.bed</em> &gt; <em>out_merged_WashU.bed</em></p>
</div></blockquote>
</li>
</ol>
</div></blockquote>
<p>Now user needs to upload the generated .bed file (<strong>Note: we do not need to convert them in gzipped / tabix formats</strong>) in the epigenome browser, by performing the following steps.</p>
<blockquote>
<div><ol class="arabic simple">
<li>Open <a class="reference external" href="http://epigenomegateway.wustl.edu/legacy/">http://epigenomegateway.wustl.edu/legacy/</a></li>
<li>Select reference genome.</li>
<li>Select <strong>Genome Browser</strong> -&gt; <strong>Apps</strong> -&gt; <strong>File upload</strong> -&gt; <strong>Choose files</strong></li>
<li>Select the generated WashU bed file and click <strong>setup</strong></li>
<li>Select Radio button for <strong>Pairwise interaction</strong> -&gt; <strong>add as Track</strong></li>
<li>Now right click on the track in the browser, and select <strong>Arc</strong>.</li>
</ol>
</div></blockquote>
</div>
</div>
<div class="section" id="visualizing-significant-interactions-in-ucsc-genome-browser">
<h2>Visualizing significant interactions in UCSC genome browser<a class="headerlink" href="#visualizing-significant-interactions-in-ucsc-genome-browser" title="Permalink to this headline">¶</a></h2>
<p>FitHiChIP loops can be converted to the bigBed format (or precisely BigInteract format) for visualizing in UCSC genome browser.
User should perform the following steps:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Check the link <a class="reference external" href="https://genome.ucsc.edu/goldenPath/help/interact.html">https://genome.ucsc.edu/goldenPath/help/interact.html</a>, example 4, for reference.</p>
</li>
<li><p class="first">Download UCSC <strong>bedToBigBed</strong> utility function, store it in system.</p>
</li>
<li><p class="first">Download the file <strong>interact.as</strong> (specified in the above mentioned link) and store it in the system.</p>
</li>
<li><p class="first">Then, to convert the FitHiChIP(L) or FitHiChIP(S) generated significant interactions (file <strong>FitHiC_Q”qval”.bed</strong>) into bed format, use the following command:</p>
<blockquote>
<div><p>awk -F’[\t]’ ‘{if (NR&gt;1) {if ($NF&gt;0) {a=int(-log($NF)/log(10)) + 1} else {a=500}; b=100; print $1”\t”(($2+$3)/2-1)”\t”(($5+$6)/2+1)”\t.\t”a”\t”b”\t.\t#0000FF\t”$1”\t”(($2+$3)/2-1)”\t”(($2+$3)/2+1)”\t.\t.\t”$4”\t”(($5+$6)/2-1)”\t”(($5+$6)/2+1)”\t.\t.”}}’ <em>FitHiC_Q”qval”.bed</em> | sort -k1,1 -k2,2n -k3,3n &gt; <em>out_FitHiChIP_converted.bed</em></p>
</div></blockquote>
</li>
<li><p class="first">Then convert the generated file into BigBed format as</p>
<blockquote>
<div><p>/path/to/bedToBigBed -as=/path/to/interact.as -type=bed5+13 <em>out_FitHiChIP_converted.bed</em> <em>ref_genome_chr.sizes</em> <em>out_loop.bb</em></p>
<p>where,</p>
<blockquote>
<div><ol class="arabic simple">
<li>/path/to/bedToBigBed: path of <strong>bedToBigBed</strong> utility in local system</li>
<li>/path/to/interact.as: path of the file <strong>interact.as</strong> in local system</li>
<li><em>ref_genome_chr.sizes</em>: Same as the parameter (B.4) (ChrSizeFile) in FitHiChIP configuration options.</li>
<li>The file <strong>out_loop.bb</strong> is the BigBed converted significant interactions. User can upload this file in UCSC genome browser for visualization.</li>
</ol>
</div></blockquote>
</div></blockquote>
</li>
<li><p class="first">To convert the output of FitHiChIP(L+M) or FitHiChIP(S+M) (file <strong>FitHiC_Q”qval”_MergeNearContacts.bed</strong>), replace “$NF” with “$9” in the command mentioned in (4). Rest of the procedure is similar.</p>
</li>
</ol>
</div></blockquote>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="DiffLoops.html" class="btn btn-neutral float-right" title="Differential analysis of HiChIP loops" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="configuration.html" class="btn btn-neutral float-left" title="Setting up configuration file" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Sourya Bhattacharyya, Ferhat Ay

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>