

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Quick start &mdash; FitHiChIP 0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../static/documentation_options.js"></script>
        <script type="text/javascript" src="../static/jquery.js"></script>
        <script type="text/javascript" src="../static/underscore.js"></script>
        <script type="text/javascript" src="../static/doctools.js"></script>
        <script type="text/javascript" src="../static/language_data.js"></script>
    
    <script type="text/javascript" src="../static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Setting up configuration file" href="configuration.html" />
    <link rel="prev" title="Installation" href="installation.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> FitHiChIP
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Quick start</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#description-of-the-test-data">Description of the test data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#configuration-parameters-according-to-the-test-data">Configuration parameters according to the test data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#sample-logs-from-the-console-corresponding-to-the-testdata">Sample logs from the console, corresponding to the TestData</a></li>
<li class="toctree-l2"><a class="reference internal" href="#details-of-output-files">Details of output files</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="configuration.html">Setting up configuration file</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Details of FitHiChIP outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffLoops.html">Differential analysis of HiChIP loops</a></li>
<li class="toctree-l1"><a class="reference internal" href="Utilities.html">Various utility scripts / functions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked question</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">FitHiChIP</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Quick start</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../sources/usage/quickstart.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="quick-start">
<h1>Quick start<a class="headerlink" href="#quick-start" title="Permalink to this headline">¶</a></h1>
<p><strong>(Assuming user has already installed the prerequisites, and is using the FitHiChIP source downloaded from GitHub)</strong></p>
<p>The file <em>sample_script.sh</em> (provided in GitHub archive) contains the sample script to execute FitHiChIP. The sample command to run FitHiChIP is:</p>
<blockquote>
<div>bash FitHiChIP_HiCPro.sh -C <em>configuration_file_name</em></div></blockquote>
<p>where,</p>
<blockquote>
<div>FitHiChIP_HiCPro.sh: main executable of FitHiChIP pipeline.</div></blockquote>
<p>In the GitHub archive, four configuration files are provided for users to check:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first"><em>configfile_BiasCorrection_CoverageBias</em>:</p>
<blockquote>
<div><ul class="simple">
<li>This configuration file can be used as template to execute FitHiChIP(L) &amp; FitHiChIP(L+M) with coverage bias regression.</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><em>configfile_BiasCorrection_ICEBias</em>:</p>
<blockquote>
<div><ul class="simple">
<li>Template of configuration file corresponding to FitHiChIP(L) &amp; FitHiChIP(L+M) with ICE bias regression.</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><em>configfile_P2P_BiasCorrection_CoverageBias</em>:</p>
<blockquote>
<div><ul class="simple">
<li>Template of configuration file corresponding to FitHiChIP(S) &amp; FitHiChIP(S+M) with coverage bias regression.</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><em>configfile_P2P_BiasCorrection_ICEBias</em>:</p>
<blockquote>
<div><ul class="simple">
<li>Template of configuration file corresponding to FitHiChIP(S) &amp; FitHiChIP(S+M) with ICE bias regression.</li>
</ul>
</div></blockquote>
</li>
</ol>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Selection of the configuration file depends on the FitHiChIP background model (i.e. loose L or stringent S) and the bias regression method (i.e. coverage bias or ICE bias) to be used.</li>
<li>User should tally these configuration files with the parameters mentioned in section <a class="reference internal" href="configuration.html#setting-up-configuration-file"><span class="std std-ref">Setting up configuration file</span></a> for understanding the configuration parameters.</li>
</ul>
</div>
<div class="section" id="description-of-the-test-data">
<h2>Description of the test data<a class="headerlink" href="#description-of-the-test-data" title="Permalink to this headline">¶</a></h2>
<p>Above mentioned configuration files are set according to the dataset provided in the folder <strong>TestData</strong> within the GitHub archieve. The folder contains sample HiChIP data, peak files, and also sample dataset for differential analysis of HiChIP loops.</p>
<p>User need to download following files within the “TestData” folder for executing FitHiChIP:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Sample_ValidPairs.txt.gz:</p>
<blockquote>
<div><p>Sample valid pairs file, an output from HiC-Pro pipeline.</p>
</div></blockquote>
</li>
<li><p class="first">Sample.Peaks.gz:</p>
<blockquote>
<div><p>Reference ChIP-seq peak information correspondin to the given data.</p>
</div></blockquote>
</li>
<li><p class="first">chrom_hg19.sizes:</p>
<blockquote>
<div><p>Size of individual chromosomes corresponding to the reference genome hg19</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>For different reference genome, user needs to download appropriate chromosome size file from the UCSC genome browser site.</p>
<p class="last">For example, above mentioned file can be downloaded from the link: &lt;<a class="reference external" href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes">http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes</a>&gt;</p>
</div>
</div></blockquote>
</li>
</ol>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The folder “TestData” also contains one directory “DiffLoopData” which contains dataset regarding differential HiChIP loop analysis.
The dataset is described in the page <a class="reference internal" href="DiffLoops.html#differential-analysis-of-hichip-loops"><span class="std std-ref">Differential analysis of HiChIP loops</span></a>.</p>
</div>
</div>
<div class="section" id="configuration-parameters-according-to-the-test-data">
<h2>Configuration parameters according to the test data<a class="headerlink" href="#configuration-parameters-according-to-the-test-data" title="Permalink to this headline">¶</a></h2>
<p>Above mentioned configuration files (tuned to the given test data) contain the following parameters:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">ValidPairs:</p>
<blockquote>
<div><p>Corresponds to the HiC-pro pipeline generated valid pairs file (gzipped or uncompressed tab delimited text format).</p>
<p>Considering the test data, path of the file “Sample_ValidPairs.txt.gz” is specified in the configuration file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>For users with no background on valid pairs file and format, we suggest to check the documentation of HiC-pro (<a class="reference external" href="https://github.com/nservant/HiC-Pro">https://github.com/nservant/HiC-Pro</a>) to get a clear undestanding of valid pairs file format.</li>
<li>FitHiChIP currently uses CIS interactions in the valid pairs file for modeling statistical significant interactions.</li>
</ul>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Currently, chromosome information within the valid pairs file need to be alphanumeric, i.e. chr1, chr2, etc. If the validpairs file contains numeric chromosome information (like 1, 2, etc), they need to be replaced with chr1, chr2, etc.</li>
</ul>
</div>
</div></blockquote>
</li>
<li><p class="first">PeakFile:</p>
<blockquote>
<div><ul class="simple">
<li>Either reference ChIP-seq peak file, or peaks inferred from HiChIP data</li>
<li>Can be bed (as in MACS2) formatted or its gzipped version.</li>
<li>Considering the test data, path of the file “Sample.Peaks.gz” is specified in the configuration file.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ul class="last simple">
<li>Chromosome information need to be alphanumeric.</li>
<li>For a detailed understanding of inferring peaks from HiChIP data, please check the page <a class="reference internal" href="Utilities.html#various-utility-scripts-functions"><span class="std std-ref">Various utility scripts / functions</span></a> and also the FitHiChIP paper.</li>
</ul>
</div>
</div></blockquote>
</li>
<li><p class="first">OutDir:</p>
<blockquote>
<div><p>Output directory which would contain all the results. Default: present working directory.</p>
</div></blockquote>
</li>
<li><p class="first">ChrSizeFile:</p>
<blockquote>
<div><p>Size of individual chromosomes corresponding to the reference genome.</p>
<p>Considering the test data, path of the file “chrom_hg19.sizes” should be specified.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Size of chromosomes corresponding to the target reference genome can be obtained from the UCSC genome browser.</p>
</div>
</div></blockquote>
</li>
<li><p class="first">IntType:</p>
<blockquote>
<div><p>Can be between 1 to 5. Values are:</p>
<blockquote>
<div><p>1 (peak-to-peak),</p>
<p>2 (peak-to-non-peak),</p>
<p>3 (peak-to-all)</p>
<p>4 (all-to-all: similar to Hi-C)</p>
<p>5 (all modes from 1 to 4 are executed)</p>
<p>Default and recommended value is 3 (peak to all foreground / output loops). Both peak to peak and peak to non-peak loops will be reported in this case.</p>
</div></blockquote>
</div></blockquote>
</li>
<li><p class="first">BINSIZE:</p>
<blockquote>
<div><p>Size of the bins (in bp). Default = 5000 (means 5 Kb).</p>
</div></blockquote>
</li>
<li><p class="first">LowDistThr:</p>
<blockquote>
<div><p>Lower distance threshold of interaction between two intervals (CIS). Default: 20000 (indicates 20 Kb). Interactions below this distance threshold will not be considered for statistical significance analysis.</p>
</div></blockquote>
</li>
<li><p class="first">UppDistThr:</p>
<blockquote>
<div><p>Upper distance threshold of interaction between two intervals (CIS). Default: 2000000 (indicates 2 Mb). Interactions above this distance threshold will not be considered for statistical significance analysis.</p>
</div></blockquote>
</li>
<li><p class="first">QVALUE:</p>
<blockquote>
<div><p>Minimum FDR (q-value) cutoff for detecting significant interactions. Default: 0.01</p>
</div></blockquote>
</li>
<li><p class="first">UseP2PBackgrnd:</p>
<blockquote>
<div><p>Two possible values 0 or 1, applicable only if IntType = 3 (peak-to-all foreground)</p>
<blockquote>
<div><p>0: FitHiChIP(L) or loose background is employed</p>
<p>1: FitHiChIP(S) or stringent background is employed.</p>
<p><strong>We recommend the value 1 for high sequencing depth, and 0 for low to moderate sequencing depth.</strong></p>
</div></blockquote>
</div></blockquote>
</li>
<li><p class="first">BiasType:</p>
<blockquote>
<div><p>Can be 1 (coverage bias) or 2 (ICE bias). <strong>We recommend 1 (and reported most of the results in our manuscript using this value).</strong></p>
</div></blockquote>
</li>
<li><p class="first">MergeInt:</p>
<blockquote>
<div><p>Boolean variable. If 1, <em>merge filtering</em> (+M) is enabled, meaning loops with adjacent bins in both sides are clubbed together via connected component modeling, to return a subset of most significant loops. We recommend 1.</p>
</div></blockquote>
</li>
<li><p class="first">OverWrite:</p>
<blockquote>
<div><p>Boolean variable. If 1, overwrites the existing output files of FitHiChIP. User may set this value 1 if previous execution was interrupted or a new version of code is being executed.</p>
</div></blockquote>
</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">For a detailed list of configuration options, please check the page <a class="reference internal" href="configuration.html#setting-up-configuration-file"><span class="std std-ref">Setting up configuration file</span></a> for a detailed description.</p>
</div>
</div></blockquote>
</div>
<div class="section" id="sample-logs-from-the-console-corresponding-to-the-testdata">
<h2>Sample logs from the console, corresponding to the TestData<a class="headerlink" href="#sample-logs-from-the-console-corresponding-to-the-testdata" title="Permalink to this headline">¶</a></h2>
<p>Upon executing the test data with one of the above mentioned configuration files, user can tally the output logs in console with the model console output provided in the file <strong>SampleConsoleOutput.txt</strong>, to debug about any errors regarding installation / execution.</p>
</div>
<div class="section" id="details-of-output-files">
<h2>Details of output files<a class="headerlink" href="#details-of-output-files" title="Permalink to this headline">¶</a></h2>
<p>Output files are generated within the specified <strong>OutDir</strong>.</p>
<p>User can check the details of FitHiChIP output files are provided in the page <a class="reference internal" href="output.html#details-of-fithichip-outputs"><span class="std std-ref">Details of FitHiChIP outputs</span></a>.</p>
<p>In particular, user should check for the following files:</p>
<blockquote>
<div><ol class="arabic simple">
<li><em>PREFIX</em>.interactions_FitHiC_Q”QVALUE”.bed</li>
<li><em>PREFIX</em>.interactions_FitHiC_Q”QVALUE”_MergeNearContacts.bed (generated only if <strong>MergeInt</strong> = 1)</li>
</ol>
</div></blockquote>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="configuration.html" class="btn btn-neutral float-right" title="Setting up configuration file" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="installation.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Sourya Bhattacharyya, Ferhat Ay

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>